### PR TITLE
#96: Added openApi for apigateway

### PR DESCRIPTION
**Pull request by mahinuzzaman**
Feat: #96 
- Create openAPI configs for aws api gateway and documentation. 
- the endpoints and schemas are divided into multiple files which will be bundled when deploying or creating documentation.
- there are placeholders in openApi files (ex. <<ENVIRONMENT>>,  <<ACCOUNT_ID>>) which will be replaced by values when deploying

In this PR I have only added OpenApi configs. the terraform config will be in a separate PR

- [x] Does not depend on other PRs
- [ ] Depends on other PR being merged first: #

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [ ] Improvement
- [x] New feature
- [ ] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?

Provide a brief overview of the changes introduced in this pull request.

## Related Issues/PRs (optional)

Link to the relevant issue(s) or provide a clear description of the problem or feature being addressed.

## How to test

- [ ] Needs regression testing (add QA needed label)

## Checklist

- [x] I have tested these changes locally and they function as expected.
- [ ] I have added or updated necessary documentation to reflect the changes.
- [ ] I have followed the project's coding style and guidelines.
- [ ] I have added appropriate test coverage to ensure the changes are functioning correctly.
- [ ] I have checked for any potential conflicts with other branches or pull requests.
- [ ] I have updated necessary OpenAPI specs

## Reviewer Request (optional)

If you have any specific requests or suggestions for reviewers, mention them here.

## Screenshots (optional)

If applicable, provide any additional screenshots or GIFs that showcase the changes made in this pull request.

## Demo Link (optional)

If applicable, provide a link to a live demo or deployment where the changes can be reviewed.

## Known Issues (optional)

If there are any known issues or limitations with the changes made in this pull request, describe them here.

## Additional Notes (optional)

Add any additional notes or comments you think are relevant to this pull request.

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

### CODE REVIEW

File: openapi/paths/auth/refresh-token.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            type: "string"
+        Access-Control-Allow-Headers:
+          schema:
+            type: "string"
+      content: {}
+  x-amazon-apigateway-request-validator: "ValidateBodyAndQuery"
+  x-amazon-apigateway-integration:
+    type: "aws"
+    httpMethod: "POST"
+    uri: "arn:aws:apigateway:<<AWS_REGION>>:lambda:path/2015-03-31/functions/arn:aws:lambda:<<AWS_REGION>>:<<ACCOUNT_ID>>:function:<<ENVIRONMENT>>-refresh-token-lambda/invocations"
```
Comment (melon-ruet) [inline openapi/paths/auth/refresh-token.yml:27]: Seems we can not publish API specs with filling up AWS information which is sensitive 🤔

File: openapi/paths/auth/register-org.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            type: "string"
+        Access-Control-Allow-Headers:
+          schema:
+            type: "string"
+      content: {}
+  x-amazon-apigateway-request-validator: "ValidateBodyAndQuery"
+  x-amazon-apigateway-integration:
+    type: "aws"
+    httpMethod: "POST"
+    uri: "arn:aws:apigateway:<<AWS_REGION>>:lambda:path/2015-03-31/functions/arn:aws:lambda:<<AWS_REGION>>:<<ACCOUNT_ID>>:function:<<ENVIRONMENT>>-register-org-lambda/invocations"
```
Comment (melon-ruet) [inline openapi/paths/auth/register-org.yml:26]: Same here. We can only apply placeholder. No real info.

**Review by melon-ruet — CHANGES_REQUESTED** [review]

File: openapi/paths/auth/refresh-token.yml (hunk 1 above)
Comment (mahinuzzaman) [reply openapi/paths/auth/refresh-token.yml:27]: yes. looking into it.

File: openapi/components/schemas/auth/RegisterOrgRequest.yml
Comment (melon-ruet) [inline openapi/components/schemas/auth/RegisterOrgRequest.yml:1]: Filename format is not consistent. Either use `PascalCase` or use use `dash-case`.
My suggestion is to use `dash-case`.

File: openapi/components/schemas/auth/RegisterOrgRequest.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+type: "object"
+required:
+  - organizationName
+properties:
+  organizationName:
```
Comment (melon-ruet) [inline openapi/components/schemas/auth/RegisterOrgRequest.yml:5]: Is it decided to use `camelCase` for query params, fields and payloads?

File: openapi/paths/auth/register-org.yml
Comment (melon-ruet) [inline openapi/paths/auth/register-org.yml:1]: Rename to `register-organization.yml`

File: openapi/.spectral.yaml
Comment (melon-ruet) [inline openapi/.spectral.yaml:1]: Mix of yml and yaml filename. Either use yaml or use yml

File: openapi/components/schemas/auth/RegisterOrgRequest.yml
Comment (melon-ruet) [reply openapi/components/schemas/auth/RegisterOrgRequest.yml:1]: How about `register-organization-payload.yml` or `register-organization-payload.yaml`

File: openapi/paths/auth/register-org.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
+post:
+  operationId: "RegisterOrg"
```
Comment (melon-ruet) [inline openapi/paths/auth/register-org.yml:2]: RegisterOrganization

**Review by melon-ruet — CHANGES_REQUESTED** [review]

File: openapi/components/schemas/auth/RegisterOrgRequest.yml (hunk 3 above)
Comment (mahinuzzaman) [reply openapi/components/schemas/auth/RegisterOrgRequest.yml:5]: yes vai. we decided to use camelCase

File: openapi/components/schemas/auth/RegisterOrgRequest.yml
Comment (mahinuzzaman) [reply openapi/components/schemas/auth/RegisterOrgRequest.yml:1]: Updated in 07f7631b18dfbd2d0f575c0372a4771d20506cc8

File: openapi/paths/auth/register-org.yml
Comment (mahinuzzaman) [reply openapi/paths/auth/register-org.yml:1]: updated in 07f7631b18dfbd2d0f575c0372a4771d20506cc8

File: openapi/.spectral.yaml
Comment (mahinuzzaman) [reply openapi/.spectral.yaml:1]: updated in 07f7631b18dfbd2d0f575c0372a4771d20506cc8

File: openapi/paths/auth/register-org.yml (hunk 4 above)
Comment (mahinuzzaman) [reply openapi/paths/auth/register-org.yml]: updated in a69ce7cbc5d389a5dc3b3f90a56ab0d457f93929

**Review by melon-ruet — APPROVED** [review]

**Pull request merged**